### PR TITLE
Deployment: Add files for docker based deployment

### PR DESCRIPTION
**Pull request by RishabhJain2018**
@deshraj Please review this PR.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 node_modules
 bower_components
 env
+media
```
Comment (deshraj) [inline .dockerignore]: Just curious, is this working properly? Did you do sanity check for this?

File: settings/staging.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from .prod import *   # noqa: ignore=F405
+
+ALLOWED_HOSTS = [
+    'staging.evalai.cloudcv.org',
+]
+
+# Database
+# https://docs.djangoproject.com/en/1.10.2/ref/settings/#databases
```
Comment (deshraj) [inline settings/staging.py]: Please remove the comments on line 7-8.

File: settings/staging.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+CORS_ORIGIN_ALLOW_ALL = False
+
+CORS_ORIGIN_WHITELIST = (
+    'evalai.cloudcv.org',
```
Comment (deshraj) [inline settings/staging.py]: We should not add this to the staging configuration, right?

File: .dockerignore (hunk 1 above)
Comment (RishabhJain2018) [reply .dockerignore]: yes! I did a check for it & it works perfectly.

**Comment by RishabhJain2018** [comment]
@deshraj Please take a look at this once & then we can merge it.

**Pull request merged**